pages/services/nifi/0.5.0-1.9.2/troubleshooting/diagnostictools/index.md

```diff
@@ -0,0 +1,153 @@
+
+From Mesos’s perspective, the Scheduler is being run as a Marathon app. Therefore we should pick marathon from this list and then find our Scheduler in the list of tasks.
+
+![scheduler](../../img/4_Scheduler.png)
+
+Figure 4. - List of active tasks
+
+Scheduler logs can be found either via the main Mesos frontpage in small clusters (possibly using the filter box at the top right), or by navigating into the list of tasks registered against the marathon framework in large clusters. In SDK services, the Scheduler is typically given the same name as the service. For example a `nifi-dev` service’s Scheduler would be named `nifi-dev`. Use the Sandbox link to view the Sandbox portion of the Scheduler filesystem, which contains files named `stdout` and `stderr`. These files respectively receive the `stdout/stderr` output of the Scheduler process, and can be examined to see what the Scheduler is doing.
+
+![stdout and stderr](../../img/5_Stderr_out.png)
+
+Figure 5. - Scheduler output
+
+## Task Logs
+
+When the issue being diagnosed has to do with the service tasks, e.g. a given task is crash looping, the task logs will likely provide more information. The tasks being run as a part of a service are registered against a framework matching the service name. Therefore, we should pick `<service-name>` from this list to view a list of tasks specific to that service.
+
+![task](../../img/6_Task.png)
+
+Figure 6. - Tasks running in a framework
+
+In the list above, we see separate lists of Active and Completed tasks:
+
+- Active tasks are still running. These give a picture of the current activity of the service.
+- Completed tasks have exited for some reason, whether successfully or due to a failure. These give a picture of recent activity of the service.
+
+<p class="message--note"><strong>NOTE: </strong>Older completed tasks will be automatically deleted and their data may no longer be available here.</p>
+
+Either or both of these lists may be useful depending on the context. Use the Sandbox link for one of these tasks and then start looking at Sandbox content. Files named `stderr` and `stdout` hold logs produced both by the SDK Executor process (a small wrapper around the service task) as well as any logs produced by the task itself. These files are automatically paginated at 2MB increments, so older logs may also be examined until they are automatically pruned.
+
+![task stderr and stdout](../../img/7_Task_stderr_out.png)
+
+Figure 7. - Output of `stderr` log file
+
+## Mesos Agent logs
+
+Occasionally, it can also be useful to examine what a given Mesos agent is doing. The Mesos Agent handles deployment of Mesos tasks to a given physical system in the cluster. One Mesos Agent runs on each system. These logs can be useful for determining if there’s a problem at the system level that is causing alerts across multiple services on that system.
```

Comment:
It's either "Mesos Agent" or "Mesos agent". Pick one.

Reply:
Mesos Agent